### ISSUE TITLE
Typecheck (vue-tsc) errors on all that is auto imported

### ISSUE THREAD

**Issue by TimGuendel** [nuxt/nuxt#28869]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v22.8.0
- Nuxt Version:     3.12.4
- CLI Version:      3.13.1
- Nitro Version:    2.9.7
- Package Manager:  pnpm@9.9.0
- Builder:          -
- User Config:      telemetry, experimental, components, nitro, imports, vite, css, devtools, debug, modules, pinia, postcss, cookieControl, runtimeConfig, compatibilityDate
- Runtime Modules:  @vueuse/nuxt@10.11.1, @pinia/nuxt@0.5.4, @pinia-plugin-persistedstate/nuxt@1.2.1, nuxt-jsonld@2.0.8, @nuxt/image@1.7.0, @zadigetvoltaire/nuxt-gtm@0.0.13, @dargmuesli/nuxt-cookie-control@8.4.10, @nuxt/test-utils/module@3.14.1, @nuxt/eslint@0.4.0, @vee-validate/nuxt@4.13.2
- Build Modules:    -

### Reproduction

Cannot reproduce.

### Describe the bug

I switched from yarn to pnpm and now I am getting errors when running typecheck (vue-tsc) - it looks like every auto imported item is throwing the same error,  example:

`Property 'ROUTES' does not exist on type 'CreateComponentPublicInstanceWithMixins<Readonly<{}> & Readonly<{}>`

I have hunderds of the same kind...Any ideas? 

### Logs

```shell-script
> nuxi typecheck

ℹ Using yup with vee-validate                                                                                                                                                                2:45:41 PM
app.vue:5:18 - error TS2339: Property 'SELECTORS' does not exist on type 'CreateComponentPublicInstanceWithMixins<ToResolvedProps<{}, {}>, { isEEA: typeof isEEA; }, {}, {}, {}, ComponentOptionsMixin, ComponentOptionsMixin, ... 18 more ..., {}>'.

[...]
```

**Comment by OrbisK** [comment]
did you try: `npx nuxi cleanup` ?

**Comment by CharlesBT** [comment]
I have the same issue that  occured recently

**Comment by felixgabler** [comment]
This has also started to happen to us. We are using bun and all of a sudden we are getting thousands of errors across the codebase when running `vue-tsc`.

I tried to pinpoint it and figured out that it started when we switched from Nuxt 3.13.0 to 3.13.1 and from Vue 3.4.38 to 3.5.3. Unfortunately, comparing the dependency trees, I cannot figure out exactly what could have lead to the issues.

The errors all look like this:

```sh
pages/page.vue:163:20 - error TS2339: Property '$t' does not exist on type 'CreateComponentPublicInstanceWithMixins<ToResolvedProps<{}, {}>, {}, {}, {}, {}, ComponentOptionsMixin, ComponentOptionsMixin, ... 18 more ..., {}>'.

163                 {{ $t('landing.educators.finalCTATitle') }}
```

```sh
pages/other-page.vue:103:24 - error TS2339: Property 'isNonNullable' does not exist on type 'CreateComponentPublicInstanceWithMixins<ToResolvedProps<{}, {}>, {}, ... 23 more ..., {}>'.

103             <div v-if="isNonNullable(saveError)" class="alert alert-danger mt-8 font-medium">
```

It seems like the typecheck does not find some of the auto-imported values (from utils or plugins) anymore.

Could this be connected? Or should I create a new issue?

**Comment by felixgabler** [comment]
Just removed vue from our `devDependencies` and it seems to have fixed it. Very curious.

**Comment by TimGuendel** [comment]
> Just removed vue from our `devDependencies` and it seems to have fixed it. Very curious.

We did not even have vue as a dependency, I just added it to see what would happen. Turns out it adds an error to the one project that uses yarn, and has no effect on the other one using pnpm 😶

**Comment by TimGuendel** [comment]
So this has nothing to do with the switch from yarn to pnpm. I updated our yarn project to the latest Nuxt version and now I am getting the same errors - all auto imported properties in vue files "do not exist"

**Comment by manniL** [comment]
did a `nuxi prepare` run?

**Comment by TimGuendel** [comment]
Yes ✅

**Comment by AndersCV** [comment]
We have the same issue in our repo as of lately. I haven't been able to pin-point the exact changes that have caused this. 

Everything that is auto-imported has lost all type definitions when referenced inside `<template>` tags: 

![image](https://github.com/user-attachments/assets/12757bb7-cfd2-4b90-ab58-3e4341a72816)

Inside `<setup>` it works like it used to: 

![image](https://github.com/user-attachments/assets/e7e70801-21ea-49bb-84cb-0bf4791cc954)

This can easily be re-created if you create a nuxt project using `npx nuxi@latest init <project-name>
` and use a auto-imported composable inside `<template>` tags

**Comment by Daviruss1969** [comment]
I had this issue for a while, just perform the two following commands fix the problem for me :
`npx nuxi cleanup`
`npx nuxi prepare`

**Comment by TimGuendel** [comment]
> This can easily be re-created if you create a nuxt project using `npx nuxi@latest init <project-name> ` and use a auto-imported composable inside `<template>` tags

Tried it, but on an empty project I am not getting this error and the typecheck works 👀

> I had this issue for a while, just perform the two following commands fix the problem for me : `npx nuxi cleanup` `npx nuxi prepare`

This does not work, as stated above.

**Comment by danielroe** [comment]
See [my comment](https://github.com/nuxt/nuxt/issues/28869#issuecomment-2337569237).

This is probably a dependency and if I can see your project or even package.json + lockfile, I can likely identify it and make a PR to fix the upstream project.

**Comment by TimGuendel** [comment]
> See [my comment](https://github.com/nuxt/nuxt/issues/28869#issuecomment-2337569237).
> 
> This is probably a dependency and if I can see your project or even package.json + lockfile, I can likely identify it and make a PR to fix the upstream project.

Dependencies:

```
    "dependencies": {
        "@highcharts/map-collection": "^2.2.0",
        "@nuxt/image": "1.7.0",
        "@pinia/nuxt": "^0.5.4",
        "@splidejs/splide": "^4.1.4",
        "@splidejs/splide-extension-auto-scroll": "^0.5.3",
        "@splidejs/splide-extension-intersection": "^0.2.0",
        "@stripe/stripe-js": "^4.1.0",
        "@types/qs": "^6.9.15",
        "@types/validator": "^13.12.1",
        "@vee-validate/nuxt": "^4.13.2",
        "@vee-validate/yup": "^4.13.2",
        "@vueuse/core": "^10.11.1",
        "change-case": "^5.4.4",
        "check-password-strength": "^2.0.10",
        "countup.js": "^2.8.0",
        "date-fns": "^3.6.0",
        "date-fns-tz": "^3.1.3",
        "dotenv": "^16.4.5",
        "fuse.js": "^7.0.0",
        "highcharts": "^11.4.6",
        "js-sha256": "^0.11.0",
        "just-clone": "^6.2.0",
        "nodemailer": "^6.9.14",
        "nuxt-jsonld": "^2.0.8",
        "pinia": "^2.1.7",
        "qs": "^6.12.3",
        "slugify": "^1.6.6",
        "stripe": "^16.10.0",
        "uuid": "^10.0.0",
        "validator": "^13.12.0",
        "xss": "^1.0.15"
    },
    "devDependencies": {
        "@dargmuesli/nuxt-cookie-control": "^8.4.6",
        "@nuxt/eslint": "^0.4.0",
        "@nuxt/test-utils": "^3.14.1",
        "@nuxtjs/eslint-config-typescript": "^12.1.0",
        "@pinia-plugin-persistedstate/nuxt": "^1.2.0",
        "@playwright/test": "^1.46.0",
        "@tailwindcss/forms": "^0.5.8",
        "@types/nodemailer": "^6.4.15",
        "@types/uuid": "^10.0.0",
        "@vue/test-utils": "^2.4.6",
        "@vueuse/nuxt": "^10.10.0",
        "@zadigetvoltaire/nuxt-gtm": "^0.0.13",
        "autoprefixer": "^10.4.19",
        "eslint": "^9.9.1",
        "eslint-config-prettier": "^9.1.0",
        "eslint-plugin-nuxt": "^4.0.0",
        "eslint-plugin-prettier": "^5.1.3",
        "eslint-plugin-vue": "^9.26.0",
        "happy-dom": "^15.7.3",
        "nuxt": "^3.13.1",
        "playwright-core": "^1.45.3",
        "postcss": "^8.4.44",
        "postcss-html": "^1.7.0",
        "prettier": "^3.3.2",
        "sass": "1.77.5",
        "stylelint": "^16.6.0",
        "stylelint-config-recommended-scss": "^14.1.0",
        "stylelint-config-recommended-vue": "^1.5.0",
        "stylelint-scss": "^6.4.1",
        "tailwindcss": "^3.4.10",
        "typescript": "^5.5.4",
        "vite-svg-loader": "^5.1.0",
        "vitest": "^2.0.4",
        "vue": "^3.5.3",
        "vue-tsc": "^2.0.26"
    },
```

Lock File:
https://gist.github.com/TimGuendel/7c8c6cdfd2054ef7faa7527b4f6fb91f

**Comment by danielroe** [comment]
@TimGuendel The issues in your case are caused by these packages:

- `nuxt-jsonld`: I raised https://github.com/ymmooot/nuxt-jsonld/pull/1443 a few weeks ago
- `@gtm-support/vue-gtm` - just update to get latest version with https://github.com/gtm-support/vue-gtm/pull/467

@AndersCV it's only `@gtm-support/vue-gtm` that is affected for you. Just update to the latest version.

**Comment by AndersCV** [comment]
@danielroe You're incredible, thank you so much for the help!

**Comment by felixgabler** [comment]
These are our dependencies

```json
"devDependencies": {
    "@aws-sdk/client-ses": "^3.645.0",
    "@aws-sdk/client-textract": "^3.645.0",
    "@dargmuesli/nuxt-cookie-control": "^8.4.10",
    "@headlessui/tailwindcss": "^0.2.1",
    "@headlessui/vue": "^1.7.23",
    "@iconify/vue": "^4.1.2",
    "@kyvg/vue3-notification": "^3.3.0",
    "@nuxt/devtools": "latest",
    "@nuxt/eslint": "^0.5.6",
    "@nuxt/scripts": "^0.8.5",
    "@nuxtjs/fontaine": "^0.4.3",
    "@nuxtjs/i18n": "^8.5.2",
    "@nuxtjs/sitemap": "^6.0.1",
    "@nuxtjs/tailwindcss": "^6.12.1",
    "@popperjs/core": "^2.11.8",
    "@sentry/node": "^8.29.0",
    "@sentry/vite-plugin": "^2.22.4",
    "@sentry/vue": "^8.29.0",
    "@supabase/sentry-js-integration": "^0.3.0",
    "@supabase/ssr": "^0.5.1",
    "@supabase/supabase-js": "^2.45.3",
    "@sxzz/eslint-config": "^4.1.6",
    "@tabler/icons-vue": "^3.15.0",
    "@tailwindcss/forms": "^0.5.9",
    "@tailwindcss/typography": "^0.5.15",
    "@tiptap-pro/extension-file-handler": "^2.11.2",
    "@tiptap/extension-bubble-menu": "^2.6.6",
    "@tiptap/extension-code-block-lowlight": "^2.6.6",
    "@tiptap/extension-color": "^2.6.6",
    "@tiptap/extension-highlight": "^2.6.6",
    "@tiptap/extension-image": "^2.6.6",
    "@tiptap/extension-link": "^2.6.6",
    "@tiptap/extension-placeholder": "^2.6.6",
    "@tiptap/extension-subscript": "^2.6.6",
    "@tiptap/extension-superscript": "^2.6.6",
    "@tiptap/extension-table": "^2.6.6",
    "@tiptap/extension-table-cell": "^2.6.6",
    "@tiptap/extension-table-header": "^2.6.6",
    "@tiptap/extension-table-row": "^2.6.6",
    "@tiptap/extension-text-align": "^2.6.6",
    "@tiptap/extension-text-style": "^2.6.6",
    "@tiptap/extension-typography": "^2.6.6",
    "@tiptap/extension-underline": "^2.6.6",
    "@tiptap/pm": "^2.6.6",
    "@tiptap/starter-kit": "^2.6.6",
    "@tiptap/vue-3": "^2.6.6",
    "@types/bun": "^1.1.8",
    "@types/common-tags": "^1.8.4",
    "@types/html-minifier": "^4.0.5",
    "@types/katex": "^0.16.7",
    "@types/lodash": "^4.17.7",
    "@types/luxon": "^3.4.2",
    "@types/markdown-it": "^14.1.2",
    "@types/mjml": "^4.7.4",
    "@types/uuid": "^10.0.0",
    "@unlok-co/nuxt-stripe": "^3.0.0",
    "@vee-validate/yup": "^4.13.2",
    "@vite-pwa/assets-generator": "^0.2.6",
    "@vite-pwa/nuxt": "^0.10.5",
    "@vueform/multiselect": "^2.6.9",
    "@vueform/slider": "^2.1.10",
    "@vuepic/vue-datepicker": "^9.0.3",
    "@vueuse/nuxt": "^11.0.3",
    "@vueuse/router": "^11.0.3",
    "camelcase-keys": "^9.1.3",
    "chart.js": "^4.4.4",
    "common-tags": "^1.8.2",
    "decamelize-keys": "^2.0.1",
    "floating-vue": "^5.2.2",
    "html-minifier": "^4.0.0",
    "katex": "^0.16.11",
    "langfuse": "^3.23.1",
    "lodash": "^4.17.21",
    "lowlight": "^3.1.0",
    "luxon": "^3.5.0",
    "markdown-it": "^14.1.0",
    "mime": "^4.0.4",
    "mini-svg-data-uri": "^1.4.4",
    "mjml": "^4.15.3",
    "nuxt": "^3.13.1",
    "openai": "^4.58.1",
    "pdfjs-dist": "^4.6.82",
    "posthog-js": "^1.161.1",
    "posthog-node": "^4.2.0",
    "prettier-plugin-tailwindcss": "^0.6.6",
    "stripe": "^16.10.0",
    "tiptap-extension-auto-joiner": "^0.1.3",
    "typescript": "^5.5.4",
    "uuid": "^10.0.0",
    "vee-validate": "^4.13.2",
    "vite-plugin-top-level-await": "^1.4.4",
    "vue-chartjs": "^5.3.1",
    "vue-component-type-helpers": "^2.1.6",
    "vue-draggable-plus": "^0.5.3",
    "vue-router": "^4.4.3",
    "vue-tsc": "^2.1.6",
    "vue3-lottie": "^3.3.1",
    "vue3-moveable": "^0.28.0",
    "yup": "^1.4.0"
  },
  "overrides": {
    "@supabase/supabase-js": "2.45.2",
    "estree-walker": "2.0.2",
    "sharp": "^0.33.3"
  }
```

The vue-tsc errors only occur if vue is listed as a dependency. If I remove it, everything works flawlessly.

**Comment by danielroe** [comment]
Please provide a reproduction so I can investigate 🙏

**Comment by felixgabler** [comment]
I tried to create one here: https://github.com/felixgabler/vue-import-repro

I think it might be connected to the git submodule for supabase-nuxt (I worked on that for a while and am still debugging some issues with the Supabase team). But I have no idea how this could mess things up around vue-tsc. Any idea?

**Comment by danielroe** [comment]
Depends on the dep range of that module. Refreshing your lockfile might be enough. Or pnpm overrides/resolutions should also work.

**Comment by xollaborator** [comment]
@danielroe thank you for engaging so much. I just found you already provided PR for a dependency that was causing the issues in my case. 

I would like to share how easy it is to verify locally if a dependency is breaking types.

To verify search for `declare module '@vue/runtime-core' {` in your `node_modules`. As you might see below, there is one match outside `vue`. After changing locally ''@vue/runtime-core'` to `'vue'` the problem was temporarily solved in my case. That confirmed a change is needed in the dependency. In my case there was already a PR but adding one for another dependency might be needed to improve the ecosystem for all of us.

![image](https://github.com/user-attachments/assets/e0072502-36bf-40d3-b052-28d414eda0ff)

**Comment by adds-bug** [comment]
@danielroe Hi, I have similar problem you have fixed and.. I'm still in it..!..

Could you give me any help? I really have terrible time for weeks from this problem..
I hope to get some present from you..

# github address
https://github.com/adds-bug/nuxt-app

# image for type error
<img width="1307" alt="스크린샷 2024-09-15 18 15 26" src="https://github.com/user-attachments/assets/ef61f379-5d2f-454a-b951-abce39ef4987">

**Comment by OsamaHaikal** [comment]
https://github.com/intlify/routing/pull/97 would this solve the issue for @adds-bug @danielroe?

**Comment by AndrewSparrowww** [comment]
> @AndersCV it's only `@gtm-support/vue-gtm` that is affected for you. Just update to the latest version.

@danielroe I have updated gtm to latest version and it fixed 90% of errors. But I have around 10 erros that makes me so confused I can't even explain how.

I have simple component which takes optional prop > and it throw same tsc error. When I make this prop required or make withDefaults > error gone. Can you advise here? It's definitely not required to use withDefaults and set prop default to null.

Added screenshot...
<img width="1103" alt="image" src="https://github.com/user-attachments/assets/eedc6e76-4621-41c0-aae4-fa121e55e2a0">

**Comment by danielroe** [comment]
@AndrewSparrowww That is an upstream Vue bug that should be fixed by https://github.com/vuejs/core/pull/11644

**Comment by danielroe** [comment]
@adds-bug your issue is due to quasar - I opened a PR last month and it should be fixed in a newer version - see https://github.com/quasarframework/quasar/pull/17448.

**Comment by yxw007** [comment]
> [@AndrewSparrowww](https://github.com/AndrewSparrowww) That is an upstream Vue bug that should be fixed by [vuejs/core#11644](https://github.com/vuejs/core/pull/11644).这是一个上游 Vue 错误，应该通过以下方式修复 [vuejs/core#11644](https://github.com/vuejs/core/pull/11644) 。
> 
> Linking [vuejs/language-tools#4832](https://github.com/vuejs/language-tools/issues/4832), [vuejs/language-tools#4704](https://github.com/vuejs/language-tools/issues/4704) and [vuejs/core#11272](https://github.com/vuejs/core/issues/11272).链接 [vuejs/language-tools#4832](https://github.com/vuejs/language-tools/issues/4832) , [vuejs/language-tools#4704](https://github.com/vuejs/language-tools/issues/4704) 和 [vuejs/core#11272](https://github.com/vuejs/core/issues/11272) 。

I found that changing the vue in.nuxt/types/imports.d.ts to @vue/runtime-core worked

![Image](https://github.com/user-attachments/assets/062c46e4-20e7-489a-a461-4d5af9231c04)

**Comment by P4sca1** [comment]
> 🚨 Vue TypeScript Changes
> For a while, in the Vue ecosystem, we've been augmenting @vue/runtime-core to add custom properties and more to vue. However, this inadvertently breaks the types for projects that augment vue - which is now the officially recommended in the docs way to augment these interfaces (for example, [ComponentCustomProperties](https://vuejs.org/api/utility-types.html#componentcustomproperties), [GlobalComponents](https://vuejs.org/guide/extras/web-components.html#web-components-and-typescript) and [so on](https://vuejs.org/guide/typescript/options-api.html#augmenting-global-properties)).
> 
> This means all libraries must update their code (or it will break the types of libraries that augment vue instead).
https://github.com/nuxt/nuxt/releases/tag/v3.13.0

**Comment by TheDutchCoder** [comment]
@P4sca1 Yeah that's not great, because now we're at the grace of authors actually updating their packages, so this can take a while before we can start using newer versions of Nuxt...

**Comment by yxw007** [comment]
@TheDutchCoder 
If you don't want to wait, you can use the solution recommended，creating a declarations.d.ts in the root of your project with the following code

```text
 ([credit](https://github.com/nuxt/nuxt/blob/v3.13.0)):

import type {
  ComponentCustomOptions as _ComponentCustomOptions,
  ComponentCustomProperties as _ComponentCustomProperties,
} from 'vue';

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties extends _ComponentCustomProperties {}
  interface ComponentCustomOptions extends _ComponentCustomOptions {}
}
```
This solution works for me

**Comment by Kasopej** [comment]
Thanks so much @danielroe  these module augmentation changes are causing lots of headaches but you've been a great help!

**Comment by fornazaric** [comment]
Hello, we're having the same problem at work with those packages:

```  "dependencies": {
    "@coders-tm/vue-number-format": "^3.34.1",
    "@nuxt/image": "^1.8.1",
    "@nuxt/kit": "^3.13.2",
    "@nuxtjs/device": "^3.1.1",
    "@nuxtjs/i18n": "^8.5.3",
    "@nuxtjs/tailwindcss": "^6.12.1",
    "@vue-leaflet/vue-leaflet": "^0.10.1",
    "@vueform/slider": "^2.1.10",
    "@vueuse/core": "^11.0.3",
    "@vueuse/nuxt": "^11.0.3",
    "class-variance-authority": "^0.7.0",
    "floating-vue": "^5.2.2",
    "leaflet": "^1.9.4",
    "luxon": "^3.5.0",
    "source-map-js": "^1.2.1",
    "tailwindcss": "^3.4.14",
    "vue": "3.5.11",
    "vue-advanced-cropper": "^2.8.9",
    "vue-i18n": "^9.14.0",
    "vue-tsc": "^2.1.6"
  },
  "devDependencies": {
    "@nuxt/devtools": "latest",
    "@nuxt/eslint": "^0.6.0",
    "@nuxt/eslint-config": "^0.6.0",
    "@nuxt/eslint-plugin": "^0.6.0",
    "@nuxt/module-builder": "^0.8.4",
    "@nuxt/schema": "^3.13.2",
    "@nuxt/test-utils": "^3.14.3",
    "@stylistic/eslint-plugin-ts": "^2.9.0",
    "@types/leaflet": "^1.9.8",
    "@types/lodash": "^4.17.10",
    "@types/luxon": "^3.4.2",
    "@types/node": "^22.7.6",
    "@vitest/coverage-v8": "^2.1.3",
    "@vitest/ui": "^2.1.3",
    "@vue/test-utils": "^2.4.5",
    "changelogen": "^0.5.5",
    "eslint": "^9.12.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-nuxt": "^4.0.0",
    "eslint-plugin-prettier": "^5.2.1",
    "eslint-plugin-vitest": "^0.5.4",
    "eslint-plugin-vue": "^9.29.0",
    "h3": "^1.13.0",
    "happy-dom": "^15.7.4",
    "nitropack": "^2.9.7",
    "nuxi": "^3.14.0",
    "nuxt": "^3.13.2",
    "prettier": "^3.2.5",
    "rollup": "^4.24.0",
    "sass": "^1.80.1",
    "typescript": "^5.6.3",
    "typescript-eslint": "^8.9.0",
    "vite": "^5.4.9",
    "vitest": "^2.1.3",
    "vitest-canvas-mock": "^0.3.3",
    "vue-router": "^4.4.5"
  }```

**Comment by keduardo2117** [comment]
It is also happening to me, it started after I updated libraries, inside tsconfig.json, VS Code was showing an error, it said it could not find the nuxt tsconfig at `  "extends": "./.nuxt/tsconfig.json",`. I looked around my project and noticed a new folder called npx and inside there was the .nuxt folder, so I updated the route to `"extends": "./npx/.nuxt/tsconfig.json",` and it fixed it.
I hope it helps you

**Comment by danielroe** [comment]
please don’t do that. this is likely caused by running a command like npx nuxi prepare npx.

**Comment by MikeBellika** [comment]
@danielroe I'm running into this error after upgrading from 3.14.0 to 3.14.1592.

The specific error is:
```
Property '$t' does not exist on type 'CreateComponentPublicInstanceWithMixins<ToResolvedProps<{ value?: number | undefined; status: AsyncDataRequestStatus; }, {}>, {}, {}, {}, {}, ComponentOptionsMixin, ... 19 more ..., {}>'.
```

**Some info about our setup**
We're running a mono repo using pnpm workspaces with shared packages in `package/*` and apps in `app/*`. In the root we got a package.json file with these dependencies:
```json
  "dependencies": {
    "@nuxt/kit": "^3.14.1592",
    "@nuxt/schema": "^3.14.1592",
    "nuxt": "^3.14.1592",
    "typescript": "^5.5.4"
  }
```
Inside one of the apps where we're getting the typescript error, we got these dependencies:
```json
  "dependencies": {
    "@mono-moxso/ui": "workspace:*",
    "@nuxt/eslint": "^0.7.0",
    "@nuxt/kit": "^3.14.1592",
    "@nuxt/schema": "^3.14.1592",
    "nuxt": "^3.14.1592",
    "@nuxtjs/i18n": "^9.1.0",
    "@nuxtjs/tailwindcss": "^6.12.0",
    "@sentry/nuxt": "^8.36.0",
    "vue": "^3.4.21",
    "vue-router": "^4.3.0",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@intlify/eslint-plugin-vue-i18n": "^3.0.0",
    "@faker-js/faker": "^8.4.1"
  }
```

**What I've tried so far**
1. Upgrading `@nuxtjs/i18n` from 8.3.1 to 9.1.0
2. Running `grep -RHl 'declare module .@vue.runtime-core' node_modules`. I get this result, which doesn't look wrong to me:
```bash
$ grep -RHl 'declare module .@vue.runtime-core' node_modules
node_modules/.pnpm/@vue+runtime-core@3.5.6/node_modules/@vue/runtime-core/dist/runtime-core.d.ts
node_modules/.pnpm/@vue+runtime-core@3.5.13/node_modules/@vue/runtime-core/dist/runtime-core.d.ts
node_modules/.pnpm/@vue+runtime-dom@3.5.12/node_modules/@vue/runtime-dom/dist/runtime-dom.d.ts
node_modules/.pnpm/@vue+runtime-core@3.5.12/node_modules/@vue/runtime-core/dist/runtime-core.d.ts
node_modules/.pnpm/@vue+runtime-dom@3.5.13/node_modules/@vue/runtime-dom/dist/runtime-dom.d.ts
node_modules/.pnpm/@vue+runtime-dom@3.5.6/node_modules/@vue/runtime-dom/dist/runtime-dom.d.ts
```
3. Removing explicit vue and vue-router dependency
4. Running `npx nuxi cleanup && npx nuxi prepare`
5. Created a new repo with only dependency on newest nuxt and @nuxtjs/i18n. Couldn't reproduce :(
6. Deleted pnpm-lock.json and node_modules and `pnpm i` from scratch.
7. Added an i18n.d.ts file in both the app and project root with this:
```typescript
import type {
  ComponentCustomOptions as _ComponentCustomOptions,
  ComponentCustomProperties as _ComponentCustomProperties,
} from 'vue';

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties extends _ComponentCustomProperties {}
  interface ComponentCustomOptions extends _ComponentCustomOptions {}
}
```

Running `pnpm ls -depth 2 | grep i18n` i get this result:
```bash
$ pnpm ls -depth 2 | grep i18n

├─┬ @nuxtjs/i18n 9.1.0
│ ├── @intlify/unplugin-vue-i18n 5.3.1
│ ├── vue-i18n 10.0.4
@nuxtjs/i18n 9.1.0
├─┬ @intlify/unplugin-vue-i18n 5.3.1
│ ├── @intlify/vue-i18n-extensions 7.0.0
│ └── vue-i18n 10.0.4 peer
├─┬ vue-i18n 10.0.4
@intlify/eslint-plugin-vue-i18n 3.0.0
```
Kind of at a loss of what to do from here. Do you have any tips to help debug or narrow down the issue?

**Comment by maxdzin** [comment]
@MikeBellika just try to remove `@intlify/eslint-plugin-vue-i18n` from the devDependencies for now, since it is not compatible with the latest @intlify/ v10 updates (https://github.com/intlify/eslint-plugin-vue-i18n/issues/566).

**Comment by ThermScissorpunch** [comment]
I am also experiencing this, I tried so many things to fix this :-(

<img width="651" alt="Image" src="https://github.com/user-attachments/assets/a9c3534b-afa9-46b2-805e-fcbb678cbbd4" />

This is my package.json:

`{
  "name": "storefront-unified-nuxt",
  "type": "module",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "multistore:dev": "NODE_TLS_REJECT_UNAUTHORIZED=0 NITRO_PORT=3000 NUXT_PUBLIC_ALOKAI_MULTISTORE_ENABLED=true yarn dev",
    "lint": "eslint .",
    "lint:fix": "eslint --fix .",
    "format": "prettier --write .",
    "start": "nuxt start",
    "typecheck": "nuxt typecheck",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "prepare": "nuxt prepare"
  },
  "dependencies": {
    "@nuxtjs/i18n": "8.1.1",
    "@storefront-ui/nuxt": "2.5.4",
    "@vsf-enterprise/cms-components-utils": "2.0.0",
    "@vue-storefront/nuxt": "6.2.0",
    "@vue-storefront/sdk": "3.3.0",
    "@vueuse/core": "10.11.0",
    "lodash-es": "4.17.21",
    "nuxt-lazy-hydrate": "1.0.0",
    "react-schemaorg": "^2.0.0",
    "schema-dts": "^1.1.2"
  },
  "devDependencies": {
    "@faker-js/faker": "8.4.1",
    "@nuxt/devtools": "1.3.8",
    "@nuxt/eslint": "1.0.1",
    "@nuxt/image": "1.7.0",
    "@nuxtjs/google-fonts": "3.2.0",
    "@storefront-ui/typography": "2.6.1",
    "@trivago/prettier-plugin-sort-imports": "4.3.0",
    "@types/lodash-es": "4.17.12",
    "@types/node": "^18.17.17",
    "@vite-pwa/nuxt": "0.5.0",
    "@vue-storefront/eslint-config": "4.1.0",
    "@vue/compiler-sfc": "3.4.31",
    "eslint": "9.14.0",
    "happy-dom": "14.12.3",
    "nuxt": "3.12.2",
    "postcss-custom-properties": "13.3.11",
    "prettier": "3.3.2",
    "sass": "1.77.6",
    "typescript": "5.4.5",
    "vue-tsc": "1.8.27"
  }
}
`

**Comment by atlesque** [comment]
> Just removed vue from our `devDependencies` and it seems to have fixed it. Very curious.

This worked for us. We had `vue` as a dependency inside a monorepo package. Removing the explicit dependency caused the auto-imports to work again.

**Comment by Pentadome** [comment]
`nuxt typecheck` deletes the entire .nuxt folder and doesnt run the code that the modules provide, and the modules of often produce code themselves,  therefore I get type errors as well

**Comment by frasza** [comment]
I have also recently came across the problem on my side with these errors (a lot of them) and I have not been able to pinpoint the core issue. I am often updating my deps, so I would say its something about that.
whenever I run "nuxi typecheck" I don't get any errors, but my IDE is full of them.

example
![Image](https://github.com/user-attachments/assets/238c8457-11e3-40c3-887f-7231f9c2fc9e)

I have augmented types as defined in examples below, I have tried downgrading multiple packages ... no luck.

@danielroe do you have any idea where to look? what else could I try? currently for me all the types and imports are broken.

**Comment by danielroe** [comment]
@frasza It sounds like it might be a Volar issue. Ensure you have hybrid mode enabled. Does that help?

**Comment by frasza** [comment]
@danielroe I had it set to "auto" before, even setting it to "true" doesn't seem to help.
I have tried different Volar versions too, no luck so far.
I guess I will have to play around with it.

PS: Sorry for direct mention here 🙏 appreciate you and your help.

**Comment by frasza** [comment]
@danielroe I have been digging hard and deep to try and understand where it breaks because hybrid on/off is so mixed,

The thing that got it finally working is -- I am using Capacitor and therefore I have "android" folder in the project. So, here is the catch! If I DELETE /android folder and run "nuxi prepare" all the IDE errors are actually gone!

But at this point I am not sure what to do. Could I somehow ignore that folder? No clue what causes that.

**Comment by danielroe** [comment]
@frasza possibly. it sounds like it's a volar/ide plugin issue rather than a Nuxt one as `vue-tsc` is working fine. see if you can tweak the configuration of Volar, try using the 'extension bisect' tool in VSCode to make sure the issue isn't caused by any other plugin, and if that doesn't work, I would advise opening an issue in https://github.com/vuejs/language-tools.

**Comment by danielroe** [comment]
I'm going to close this issue as it's been used to cover _any number_ of unrelated issues, from IDE issues (ie. replicable only in IDE and _not_ in vue-tsc) to issues in downstream libraries (which augment the wrong library, like `@vue/runtime-dom`).

If you encounter an issue like this, there are some remedies you can try in the comments.

But if none of them work, please open a new issue _with_ a reproduction and I will very happily help you debug the source of the problem.
